met-web/src/components/common/Modals/DayCalculator/index.tsx

```diff
@@ -0,0 +1,703 @@
+import { When } from 'react-if';
+import dayjs from 'dayjs';
+import { muitheme, options } from './constants';
+import DayZeroRules from './dayZeroRules';
+
+interface DayCalcModalProps {
+    open: boolean;
+    updateModal: (open: boolean) => void;
+}
+interface ISelectOptions {
+    id: number;
+    label: string;
+}
+
+const DayCalculatorModal = ({ open, updateModal }: DayCalcModalProps) => {
+    const isSmallScreen = useMediaQuery(muitheme.breakpoints.down('lg'));
+
+    // variable to capture the option selected within Calculation Type
+    const [selectedOption, setSelectedOption] = useState<ISelectOptions | null>(options[0]);
+
+    // variable to throw error in case suspension or resumption dates are blank
+    const [dayCalcError, setdayCalcError] = useState({
+        suspensionDate: false,
+        resumptionDate: false,
+    });
+    const validateDayCalcInputs = () => {
+        const errors = {
+            suspensionDate: !suspensionDate,
+            resumptionDate: !resumptionDate,
+        };
+        setdayCalcError(errors);
+
+        return Object.values(errors).some((isError: unknown) => isError);
+    };
+
+    // variable to show or hide the day zero rules
+    const [showHideStatus, setShowHideStatus] = useState(false);
+
+    // variable to capture changes on the input fields
+    const [dayCalcData, setDayCalcData] = useState({
+        startDate: '',
+        endDate: '',
+        suspensionDate: '',
+        resumptionDate: '',
+        numberOfDays: 0,
+    });
+    const { startDate, endDate, suspensionDate, resumptionDate, numberOfDays } = dayCalcData;
+    const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+        setDayCalcData({
+            ...dayCalcData,
+            [e.target.name]: e.target.value,
+        });
+        setdayCalcError({
+            ...dayCalcError,
+            [e.target.name]: false,
+        });
+    };
+
+    // reset to default
+    const resetToDefault = () => {
+        setDayCalcData({
+            startDate: '',
+            endDate: '',
+            suspensionDate: '',
+            resumptionDate: '',
+            numberOfDays: 0,
+        });
+    };
+
+    // calculator
+    const calculator = () => {
+        let calcStartDate = startDate;
+        let calcEndDate = endDate;
+        let calcNumberOfDays = numberOfDays;
+        if (selectedOption?.label == options[0].label) {
+            if (startDate && endDate && !numberOfDays) {
+                calcNumberOfDays = dayjs(endDate).diff(dayjs(startDate), 'days');
+            } else if (startDate && !endDate && numberOfDays) {
+                calcEndDate = dayjs(startDate).add(numberOfDays, 'day').format('YYYY-MM-DD');
+            } else if (!startDate && endDate && numberOfDays) {
+                calcStartDate = dayjs(endDate).subtract(numberOfDays, 'day').format('YYYY-MM-DD');
+            }
+        }
+
+        if (selectedOption?.label == options[1].label) {
```

Comment:
Please write enough comments about the logic ;
One of the better style is to use meaningful variable names 
https://www.loadingdeveloper.com/writing-cleaner-code-making-cleaner-if-statements/

Reply:
updated as per review comments